src/main/java/de/tum/in/www1/artemis/domain/ShortAnswerQuestion.java

```diff
@@ -122,44 +120,317 @@ public void setCorrectMappings(List<ShortAnswerMapping> shortAnswerMappings) {
     // jhipster-needle-entity-add-getters-setters - JHipster will add getters and setters here, do not remove
 
     @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-        ShortAnswerQuestion shortAnswerQuestion = (ShortAnswerQuestion) o;
-        if (shortAnswerQuestion.getId() == null || getId() == null) {
+    public Boolean isValid() {
+        // check general validity (using superclass)
+        if (!super.isValid()) {
             return false;
         }
-        return Objects.equals(getId(), shortAnswerQuestion.getId());
-    }
 
-    @Override
-    public int hashCode() {
-        return Objects.hashCode(getId());
-    }
+        // check if at least one correct mapping exists
+        return getCorrectMappings() != null && !getCorrectMappings().isEmpty();
 
-    @Override
-    public String toString() {
-        return "ShortAnswerQuestion{" +
-            "id=" + getId() +
-            "}";
+        // TODO (?): Add checks for "is solvable" and "no misleading correct mapping" --> look at the implementation in the client
     }
 
+     /**
+      * Get all solution items that are mapped to the given spot
+      *
+      * @param spot the spot  we want to find the correct solutions for
+      * @return all solutions that are defined as correct for this spot
+      */
+     public Set<ShortAnswerSolution> getCorrectSolutionForSpot(ShortAnswerSpot spot) {
+         Set<ShortAnswerSolution> result = new HashSet<>();
+         for (ShortAnswerMapping mapping : correctMappings) {
+             if (mapping.getSpot().equals(spot)) {
+                 result.add(mapping.getSolution());
+             }
+         }
+         return result;
+     }
+
+     /**
+      * Get solution by ID
+      *
+      * @param solutionId the ID of the solution, which should be found
+      * @return the sollution with the given ID, or null if the solution is not contained in this question
+      */
+     public ShortAnswerSolution findSolutionById (Long solutionId) {
+
+         if (solutionId != null) {
```

Comment:
I suggest to do the other way around, it will save you one indentation for a big part of the body of the method - while in this case it doesn't do much difference, it is a good practice in general.

```
public ShortAnswerSolution findSolutionById (Long solutionId) {
    if (solutionId == null) {
        return null;
    } 

    for (ShortAnswerSolution solution : solutions) {
        // return dragItem if the IDs are equal
        if (solution.getId().equals(solutionId)) {
            return solution;
         }
     }
}
```

Reply:
There was still a return needed in the end as the for loop may not return a value (solutions is empty). 
I think that the indentation was not that bad and now it may be confusing as there are three returns now and two of them with null. But if it is good practice I can keep it that way :)

```
public ShortAnswerSolution findSolutionById (Long solutionId) {
        if (solutionId == null){
            return null;
        }
        // iterate through all solutions of this quiz
         for (ShortAnswerSolution solution : solutions) {
             // return solution if the IDs are equal
             if (solution.getId().equals(solutionId)) {
                 return solution;
             }
         }
         return null;
     }
```

Reply:
After discussion with Riccardo, we decided to leave the code like it was, as in this case it does not improve much. :)